src/projections.F90

```diff
@@ -488,7 +490,25 @@ subroutine point_volume(x0, tu, tv, tw, ku, kv, kw, coef, nctlu, nctlv, nctlw, n
             end if
         end do
 
-        if (NORM2(grad) < eps) then
+        ! for convergence, we used to check if ||grad||_2 was less than eps.
+        ! This tells us that the search cannot make much more progress;
+        ! we found a direct projection so R is very small.
+        ! However, this approach has a pitfall when the b-spline volume itself
+        ! is very small in size in one or more directions.
+        ! When the volume is small, its derivative will also be small.
+        ! As a result, the gradient vector, which is R dot deriv, will appear small.
+        ! The default tolerances for the newton convergence here is 1e-12
+        ! and the default projection tolerance we use in pygeo is 1e-10.
+        ! If the FFD derivative is on the order of 1e-2 or less,
+        ! the newton will quit when eps is about 1e-12, but R can still be above 1e-10.
+        ! The newton solver quits successfully in this case,
+        ! however, the higher level point embedding routine thinks
+        ! we are not inside the b-spline volume since we could not reach the
+        ! 1e-10 tolerance. instead, we just check if we are within the
+        ! specified tolerance in physical coordinates.
+        ! if the point is physically outside the b-spline volume, the line search fail
+        ! check below will catch that case and terminate early.
```

Comment:
We don't have the line search fail check anymore right? Can you rephrase this part?

Reply:
I rephrased it. It was actually unnecessarily long so I simplified it.